docs/manual/philosophy.md

```diff
@@ -0,0 +1,172 @@
+---
+layout: default
+title: rpm.org - RPM's Philosophy
+---
+# RPM's Philosophy
+
+RPM is a general purpose software manager. It has
+two major parts: rpmbuild that combines upstream sources with a meta
+data and build instructions in a "spec file" and additional files and
+patches. It allows to bundle all this up into a source package and
+build the software into a binary package. The second part - the rpm
+utility - can then install, update or remove the packages after
+checking for consistency.
+
+RPM differs from special purpose package managers - like those
+targeting a specific programming language - in trying to not make
+assumptions about how software looks like or gets packaged. Packaging
+software can be messy and RPM accommodates for that.
+
+It still offers help to create a POSIX like operation system. Macros
+parameterize locations to make the build instructions more generic and
+more easily transferable between distributions. Nevertheless RPM
+packages are targeting a specific distribution (and release thereof). They are
+typically not suited to be installed or even built elsewhere without
+tweaking. RPM acknowledges that different distributions have
+different goals, resulting in different design decisions. The
+specifics of the packages often reflect these.
+
+RPM as a upstream project still tries to keep distributions from
+diverging unnecessarily but is aware that these differences are the
+reason distributions exist in the first place.
+
+## RPM in the Software handling stack
+
+RPM sees itself as filling specific layers in the software handling
+stack. It relies on build systems like make and the upstream provided
+build scripts to orchestrate the actual build and passes the acquiring
+and selection of the "right" packages up to updaters like yum, dnf,
+zypper, etc.
+
+The stack typically looks like this:
+
+ * Upstream build scripts using tools like maker, cmake, ant, ...
+ * rpmbuild for running those via a Spec file
+ * Build systems for installing build dependencies and keeping track of build artifacts
+ * Package repositories offering binary packages
+ * Installers/Updaters selecting and downloading packages
+ * rpm checking for consistency and installing/updating the packages
+
+As such RPM is only a part in a larger set of tools and some users may
+never actually interact with RPM directly.
+
+## Design goals
+
+* General purpose software management
+* Building from pristine sources
+* Unattended operation
+  * for both building
```

Comment:
"for both building" doesn't make sense, looks like an editing leftover :)

Reply:
Fixed